modules/core/src/utils/deep-equal.ts

```diff
@@ -1,17 +1,17 @@
 // Partial deep equal (only recursive on arrays)
-export function deepEqual(a: any, b: any): boolean {
+export function deepEqual(a: any, b: any, recursive: boolean = false): boolean {
```

Comment:
Nit: It was a bit weird to have a function named `deepEqual` that compares one level, but doesn't recurse. Worse, it recursed on some things but not others. 

While adding the option is reasonable, there could perhaps be some opportunity to create a better named function or set of functions here, or at least add proper JSDoc.